### PR TITLE
Unify SLOG

### PR DESCRIPTION
**Pull request by lynxlynxlynx**
#1487

most of this was regexed. One or two improvements on the way there. Probably still some redundant c_str() calls around, as the logging statements are often split into several lines, sometimes even out of context (4th+ line). One addition of SGPSector::FromSectorID which I suspect was just logging wrong before.

### CODE REVIEW

File: src/sgp/SGP.cc [hunk 1]
```diff
@@ -374,7 +374,7 @@ int main(int argc, char* argv[])
 			Logger_initialize(tempFilename.c_str());
 			for (const ST::string& msg : problems)
 			{
-				SLOGW("%s", msg.c_str());
+				SLOGW("{}", msg);
```
Comment (ja2-derek) [inline src/sgp/SGP.cc:377]: Maybe we can just do `SLOGW(msg)`

File: src/sgp/FileMan.cc [hunk 2]
```diff
@@ -57,7 +57,7 @@ uint64_t FileMan::getFreeSpace(const ST::string& path)
 	if (!Fs_freeSpace(path.c_str(), &bytes))
 	{
 		RustPointer<char> err(getRustError());
-		STLOGW("FileMan::getFreeSpace('{}') failed: %s", path, err.get());
+		SLOGW("FileMan::getFreeSpace('{}') failed: %s", path, err.get());
```
Comment (ja2-derek) [inline src/sgp/FileMan.cc:60]: Looks like there is an error in the original format string. It should be replaced with `{}` too.

File: src/sgp/SGP.cc (hunk 1 above)
Comment (lynxlynxlynx) [reply src/sgp/SGP.cc:377]: I actually tried, it errors out.

File: src/externalized/Soldier.cc [hunk 3]
```diff
@@ -44,7 +44,7 @@ void Soldier::removePendingAction()
 {
 	if(mSoldier->ubPendingAction != NO_PENDING_ACTION)
 	{
-		SLOGD("%s: remove pending action %s",
+		SLOGD("{}: remove pending action {}",
 			getPofileName().c_str(),
```
Comment (selaux) [inline src/externalized/Soldier.cc:48]: `.c_str()` can probably be removed.

File: src/externalized/Soldier.cc [hunk 4]
```diff
@@ -90,7 +90,7 @@ bool Soldier::anyoneHasPendingAction(UINT8 action, UINT8 team)
 
 void Soldier::setPendingAction(UINT8 action)
 {
-	SLOGD("%s: set pending action %s (previous %s)",
+	SLOGD("{}: set pending action {} (previous {})",
 		getPofileName().c_str(),
```
Comment (selaux) [inline src/externalized/Soldier.cc:94]: `.c_str()` can probably be removed.

File: src/game/Tactical/Morale.cc [hunk 5]
```diff
@@ -406,11 +406,11 @@ void HandleMoraleEvent(SOLDIERTYPE *pSoldier, INT8 bMoraleEvent, const SGPSector
...
 	if (pSoldier == NULL)
 	{
-		STLOGD("Handling morale event {} at {}", bMoraleEvent, sMap.AsLongString());
+		SLOGD("Handling morale event {} at {}", bMoraleEvent, sMap.AsLongString());
 	}
 	else
 	{
-		STLOGD("Handling morale event {} for {} at {}",
+		SLOGD("Handling morale event {} for {} at {}",
 						 bMoraleEvent, pSoldier->name.c_str(), sMap.AsLongString());
```
Comment (selaux) [inline src/game/Tactical/Morale.cc:414]: `.c_str()` can probably be removed.

File: src/game/Tactical/OppList.cc [hunk 6]
```diff
@@ -1234,7 +1234,7 @@ static INT16 ManLooksForMan(SOLDIERTYPE* pSoldier, SOLDIERTYPE* pOpponent, UINT8
 
 static void ManLooksForOtherTeams(SOLDIERTYPE* pSoldier)
 {
-	SLOGD("MANLOOKSFOROTHERTEAMS ID %d(%s) team %d side %d",
+	SLOGD("MANLOOKSFOROTHERTEAMS ID {}({}) team {} side {}",
 		pSoldier->ubID, pSoldier->name.c_str(), pSoldier->bTeam, pSoldier->bSide);
```
Comment (selaux) [inline src/game/Tactical/OppList.cc:1238]: `.c_str()` can probably be removed.

File: src/game/Tactical/OppList.cc [hunk 7]
```diff
@@ -1308,7 +1308,7 @@ static void HandleManNoLongerSeen(SOLDIERTYPE* pSoldier, SOLDIERTYPE* pOpponent,
 		// THIS MUST HAPPEN EVEN FOR ENEMIES, TO MAKE THEIR PUBLIC opplist DECAY!
 		if (TeamNoLongerSeesMan(pSoldier->bTeam, pOpponent, pSoldier, 0))
 		{
-			SLOGD("TeamNoLongerSeesMan: ID %d(%s) to ID %d",
+			SLOGD("TeamNoLongerSeesMan: ID {}({}) to ID {}",
 				pSoldier->ubID, pSoldier->name.c_str(), pOpponent->ubID);
```
Comment (selaux) [inline src/game/Tactical/OppList.cc:1312]: `.c_str()` can probably be removed.

File: src/game/Tactical/OppList.cc [hunk 8]
```diff
@@ -1328,7 +1328,7 @@ static void HandleManNoLongerSeen(SOLDIERTYPE* pSoldier, SOLDIERTYPE* pOpponent,
 		}
 	}
 	else
-		SLOGD("ManLooksForMan: ID %d(%s) to ID %d Personally seen, public %d",
+		SLOGD("ManLooksForMan: ID {}({}) to ID {} Personally seen, public {}",
 			pSoldier->ubID, pSoldier->name.c_str(), pOpponent->ubID, *pbPublOL);
```
Comment (selaux) [inline src/game/Tactical/OppList.cc:1332]: `.c_str()` can probably be removed.

File: src/game/Tactical/OppList.cc [hunk 9]
```diff
@@ -1364,14 +1364,14 @@ static INT16 ManLooksForMan(SOLDIERTYPE* pSoldier, SOLDIERTYPE* pOpponent, UINT8
...
+		SLOGD("ManLooksForMan - Looking for self - ID {}({})", pSoldier->ubID, pSoldier->name);
 		return(FALSE);
 	}
 
 	// if we're somehow looking while inactive, at base, dead or dying
 	if (!pSoldier->bActive || !pSoldier->bInSector || (pSoldier->bLife < OKLIFE))
 	{
-		SLOGD("ManLooksForMan - WE are inactive/dead etc ID %d(%s)to ID %d",
+		SLOGD("ManLooksForMan - WE are inactive/dead etc ID {}({})to ID {}",
 			pSoldier->ubID, pSoldier->name.c_str(), pOpponent->ubID);
```
Comment (selaux) [inline src/game/Tactical/OppList.cc:1375]: `.c_str()` can probably be removed.

File: src/game/Tactical/OppList.cc [hunk 10]
```diff
@@ -1389,7 +1389,7 @@ static INT16 ManLooksForMan(SOLDIERTYPE* pSoldier, SOLDIERTYPE* pOpponent, UINT8
 	// if he's looking for a guy who is on the same team
 	if (pSoldier->bTeam == pOpponent->bTeam)
 	{
-		SLOGD("ManLooksForMan - SAME TEAM ID %d(%s)to ID %d",
+		SLOGD("ManLooksForMan - SAME TEAM ID {}({})to ID {}",
 			pSoldier->ubID, pSoldier->name.c_str(), pOpponent->ubID);
```
Comment (selaux) [inline src/game/Tactical/OppList.cc:1393]: `.c_str()` can probably be removed.

File: src/game/Tactical/OppList.cc [hunk 11]
```diff
@@ -1471,7 +1471,7 @@ static INT16 ManLooksForMan(SOLDIERTYPE* pSoldier, SOLDIERTYPE* pOpponent, UINT8
 
 	// calculate how many spaces away soldier is (using Pythagoras' theorem)
 	sDistAway = PythSpacesAway(pSoldier->sGridNo,pOpponent->sGridNo);
-	SLOGD("MANLOOKSFORMAN: ID %d(%s) to ID %d: sDistAway %d sDistVisible %d",
+	SLOGD("MANLOOKSFORMAN: ID {}({}) to ID {}: sDistAway {} sDistVisible {}",
 		pSoldier->ubID, pSoldier->name.c_str(), pOpponent->ubID, sDistAway, sDistVisible);
```
Comment (selaux) [inline src/game/Tactical/OppList.cc:1475]: `.c_str()` can probably be removed.

File: src/game/Tactical/OppList.cc [hunk 12]
```diff
@@ -1486,7 +1486,7 @@ static INT16 ManLooksForMan(SOLDIERTYPE* pSoldier, SOLDIERTYPE* pOpponent, UINT8
 		}
 		else
 		{
-			SLOGD("FAILED LINEOFSIGHT: ID %d (%s)to ID %d Personally %d, public %d",
+			SLOGD("FAILED LINEOFSIGHT: ID {} ({})to ID {} Personally {}, public {}",
 				pSoldier->ubID, pSoldier->name.c_str(), pOpponent->ubID, *pPersOL, *pbPublOL);
```
Comment (selaux) [inline src/game/Tactical/OppList.cc:1490]: `.c_str()` can probably be removed.

File: src/game/Tactical/OppList.cc [hunk 13]
```diff
@@ -1797,7 +1797,7 @@ static void ManSeesMan(SOLDIERTYPE& s, SOLDIERTYPE& opponent, UINT8 const caller
 			s.bSide != opponent.bSide)
 		{
 			AddOneOpponent(&s);
-			SLOGD("ManSeesMan: ID %d(%s) to ID %d NEW TO ME",
+			SLOGD("ManSeesMan: ID {}({}) to ID {} NEW TO ME",
 				s.ubID, s.name.c_str(), opponent.ubID);
```
Comment (selaux) [inline src/game/Tactical/OppList.cc:1801]: `.c_str()` can probably be removed.

File: src/game/Tactical/OppList.cc [hunk 14]
```diff
@@ -1852,7 +1852,7 @@ static void ManSeesMan(SOLDIERTYPE& s, SOLDIERTYPE& opponent, UINT8 const caller
 	}
 	else
 	{
-		SLOGD("ManSeesMan: ID %d(%s) to ID %d ALREADYSEENCURRENTLY",
+		SLOGD("ManSeesMan: ID {}({}) to ID {} ALREADYSEENCURRENTLY",
 			s.ubID, s.name.c_str(), opponent.ubID);
```
Comment (selaux) [inline src/game/Tactical/OppList.cc:1856]: `.c_str()` can probably be removed.

File: src/game/Tactical/OppList.cc [hunk 15]
```diff
@@ -1930,7 +1930,7 @@ static void ManSeesMan(SOLDIERTYPE& s, SOLDIERTYPE& opponent, UINT8 const caller
 				opponent.pLevelNode->ubShadeLevel = opponent.ubFadeLevel;
 			}
 		}
-		SLOGD("ID %d (%s) MAKING %d VISIBLE",
+		SLOGD("ID {} ({}) MAKING {} VISIBLE",
 			s.ubID, s.name.c_str(), opponent.ubID);
```
Comment (selaux) [inline src/game/Tactical/OppList.cc:1934]: `.c_str()` can probably be removed.

File: src/game/Tactical/OppList.cc [hunk 16]
```diff
@@ -1984,7 +1984,7 @@ static void OtherTeamsLookForMan(SOLDIERTYPE* pOpponent)
 		// assume he's no longer visible, until one of our mercs sees him again
 		pOpponent->bVisible = 0;
 	}
-		SLOGD("OTHERTEAMSLOOKFORMAN ID %d(%s) team %d side %d",
+		SLOGD("OTHERTEAMSLOOKFORMAN ID {}({}) team {} side {}",
 			pOpponent->ubID, pOpponent->name.c_str(), pOpponent->bTeam, pOpponent->bSide);
```
Comment (selaux) [inline src/game/Tactical/OppList.cc:1988]: `.c_str()` can probably be removed.

File: src/game/Tactical/OppList.cc [hunk 17]
```diff
@@ -2089,7 +2089,7 @@ static void RemoveOneOpponent(SOLDIERTYPE* pSoldier)
 
 	if ( pSoldier->bOppCnt < 0 )
 	{
-		SLOGD("Oppcnt for %d (%s) tried to go below 0",
+		SLOGD("Oppcnt for {} ({}) tried to go below 0",
 			pSoldier->ubID, pSoldier->name.c_str());
```
Comment (selaux) [inline src/game/Tactical/OppList.cc:2093]: `.c_str()` can probably be removed.

File: src/game/Tactical/OppList.cc [hunk 18]
```diff
@@ -4283,7 +4283,7 @@ static void TellPlayerAboutNoise(SOLDIERTYPE* const s, SOLDIERTYPE const* const
 
 	if (noise_maker && s->bTeam == OUR_TEAM && s->bTeam == noise_maker->bTeam)
 	{
-		SLOGE("%s (%d) heard noise from %s (%d), noise at %dL%d, type %d",
+		SLOGE("{} ({}) heard noise from {} ({}), noise at {}L{}, type {}",
 			s->name.c_str(), s->ubID, noise_maker->name.c_str(), noise_maker->ubID, sGridNo, level, noise_type);
```
Comment (selaux) [inline src/game/Tactical/OppList.cc:4287]: `.c_str()` can probably be removed.

File: src/game/TacticalAI/AIMain.cc [hunk 19]
```diff
@@ -1078,7 +1078,7 @@ static void TurnBasedHandleNPCAI(SOLDIERTYPE* pSoldier)
 	}
 
 
-	SLOGD("HandleManAI - DECIDING for guynum %d(%s) at gridno %d, APs %d",
+	SLOGD("HandleManAI - DECIDING for guynum {}({}) at gridno {}, APs {}",
 		pSoldier->ubID, pSoldier->name.c_str(), pSoldier->sGridNo, pSoldier->bActionPoints);
```
Comment (selaux) [inline src/game/TacticalAI/AIMain.cc:1082]: `.c_str()` can probably be removed.

File: src/game/TacticalAI/AIMain.cc [hunk 20]
```diff
@@ -1350,7 +1350,7 @@ INT8 ExecuteAction(SOLDIERTYPE *pSoldier)
 				RESETTIMECOUNTER( pSoldier->AICounter, pSoldier->usActionData );
 				if (pSoldier->ubProfile != NO_PROFILE)
 				{
-					SLOGD("%s waiting %d from %d",
+					SLOGD("{} waiting {} from {}",
 								pSoldier->name.c_str(), pSoldier->AICounter, GetJA2Clock());
```
Comment (selaux) [inline src/game/TacticalAI/AIMain.cc:1354]: `.c_str()` can probably be removed.

File: src/game/TacticalAI/Movement.cc [hunk 21]
```diff
@@ -290,7 +290,7 @@ INT8 RandomPointPatrolAI(SOLDIERTYPE *pSoldier)
 
 
 	// passed all tests - start moving towards next patrol point
-	SLOGD("%s - POINT PATROL to grid %d",
+	SLOGD("{} - POINT PATROL to grid {}",
 		pSoldier->name.c_str(), pSoldier->usActionData);
```
Comment (selaux) [inline src/game/TacticalAI/Movement.cc:294]: `.c_str()` can probably be removed.

File: src/game/TacticalAI/NPC.cc [hunk 22]
```diff
@@ -1135,7 +1135,7 @@ static UINT8 NPCConsiderQuote(UINT8 const ubNPC, UINT8 const ubMerc, Approach co
 
 	if (ubApproach != NPC_INITIATING_CONV && ubMerc != NO_PROFILE)
 	{
-		SLOGD("New Approach for NPC ID: %d '%s' against Merc: %d '%s'\n\
+		SLOGD("New Approach for NPC ID: {} '{}' against Merc: {} '{}'\n\
 														\tTesting Record #: %d",
 					ubNPC, GetProfile(ubNPC).zNickname.c_str(), ubMerc, GetProfile(ubMerc).zNickname.c_str(), ubQuoteNum);
```
Comment (selaux) [inline src/game/TacticalAI/NPC.cc:1140]: `.c_str()` can probably be removed.

File: src/game/TacticalAI/NPC.cc [hunk 23]
```diff
@@ -1187,7 +1187,7 @@ static UINT8 NPCConsiderQuote(UINT8 const ubNPC, UINT8 const ubMerc, Approach co
 		fTrue = CheckFact((Fact)pNPCQuoteInfo->usFactMustBeTrue, ubNPC);
 		if (ubApproach != NPC_INITIATING_CONV)
 		{
-			SLOGD("Fact (%d:'%s') Must be True, status is %s",
+			SLOGD("Fact ({}:'{}') Must be True, status is {}",
 						pNPCQuoteInfo->usFactMustBeTrue, FactDescText[pNPCQuoteInfo->usFactMustBeTrue].c_str(),
```
Comment (selaux) [inline src/game/TacticalAI/NPC.cc:1191]: `.c_str()` can probably be removed.

File: src/game/TacticalAI/NPC.cc [hunk 24]
```diff
@@ -1199,7 +1199,7 @@ static UINT8 NPCConsiderQuote(UINT8 const ubNPC, UINT8 const ubMerc, Approach co
 		fTrue = CheckFact((Fact)pNPCQuoteInfo->usFactMustBeFalse, ubNPC);
 		if (ubApproach != NPC_INITIATING_CONV)
 		{
-			SLOGD("Fact(%d:'%s') Must be False status is  %s",
+			SLOGD("Fact({}:'{}') Must be False status is  {}",
 						pNPCQuoteInfo->usFactMustBeFalse, FactDescText[pNPCQuoteInfo->usFactMustBeFalse].c_str(),
```
Comment (selaux) [inline src/game/TacticalAI/NPC.cc:1203]: `.c_str()` can probably be removed.

File: src/game/TacticalAI/NPC.cc [hunk 25]
```diff
@@ -1608,7 +1608,7 @@ void ConverseFull(UINT8 const ubNPC, UINT8 const ubMerc, Approach bApproach, UIN
 					break;
 				case TRIGGER_NPC:
 					// if triggering, pass in the approach data as the record to consider
-					STLOGD("Handling trigger {}/{} at {}", gMercProfiles[ubNPC].zNickname.c_str(), approach_record, GetJA2Clock());
+					SLOGD("Handling trigger {}/{} at {}", gMercProfiles[ubNPC].zNickname.c_str(), approach_record, GetJA2Clock());
```
Comment (selaux) [inline src/game/TacticalAI/NPC.cc:1611]: `.c_str()` can probably be removed.

File: src/game/TacticalAI/PanicButtons.cc [hunk 26]
```diff
@@ -319,7 +319,7 @@ INT8 PanicAI(SOLDIERTYPE *pSoldier, UINT8 ubCanMove)
 					{
 						// blow up the all the PANIC bombs (or just the journal)
 						pSoldier->usActionData = sPanicTriggerGridNo;
-						SLOGD("%s pulls panic trigger at grid %d",
+						SLOGD("{} pulls panic trigger at grid {}",
 									pSoldier->name.c_str(), pSoldier->usActionData);
```
Comment (selaux) [inline src/game/TacticalAI/PanicButtons.cc:323]: `.c_str()` can probably be removed.

File: src/game/TileEngine/WorldMan.cc [hunk 27]
```diff
@@ -1126,7 +1126,7 @@ BOOLEAN AddMercStructureInfoFromAnimSurface(const INT16 sGridNo, SOLDIERTYPE* co
 	bool const success = AddStructureToWorld(sGridNo, s->bLevel, sr, n);
 	if (!success)
 	{
-		SLOGD("add struct info for merc %d (%s), at %d direction %d failed",
+		SLOGD("add struct info for merc {} ({}), at {} direction {} failed",
 					s->ubID, s->name.c_str(), sGridNo, s->bDirection);
```
Comment (selaux) [inline src/game/TileEngine/WorldMan.cc:1130]: `.c_str()` can probably be removed.

**Review by selaux — COMMENTED** [review]
Sorry for the amount of comments. They are cases where `.c_str()` can still be removed.

**Comment by lynxlynxlynx** [comment]
Ok, this is mergable now. :)

File: src/externalized/Soldier.cc [hunk 28]
```diff
@@ -45,7 +45,7 @@ void Soldier::removePendingAction()
 	if(mSoldier->ubPendingAction != NO_PENDING_ACTION)
 	{
 		SLOGD("{}: remove pending action {}",
-			getPofileName().c_str(),
```
Comment (ja2-derek) [inline src/externalized/Soldier.cc:48]: This typo hurts my eyes...

**Comment by ja2-derek** [comment]
LGTM!

**Pull request merged**